src/core/hle/kernel/thread.h

```diff
@@ -568,9 +570,62 @@ class Thread final : public SynchronizationObject {
         return has_exited;
     }
 
+    struct QueueEntry {
+    private:
+        Thread* prev;
+        Thread* next;
+
+    public:
+        constexpr QueueEntry() : prev(nullptr), next(nullptr) { /* ... */
+        }
+
+        constexpr void Initialize() {
+            this->prev = nullptr;
+            this->next = nullptr;
+        }
+
+        constexpr Thread* GetPrev() const {
+            return this->prev;
+        }
+        constexpr Thread* GetNext() const {
+            return this->next;
+        }
+        constexpr void SetPrev(Thread* t) {
```

Comment:
```suggestion
        constexpr void SetPrev(Thread* thread) {
```

So IDEs show a meaningful name with parameter inspection